### ISSUE TITLE
no-global-function-names does not allow using CSS math functions

### ISSUE THREAD

**Issue by in-in** [stylelint-scss/stylelint-scss#486]
This code causes the error `Expected math.min instead of min   scss/no-global-function-names`

```scss
.foo {
  width: min(300px, 30%);
}
```

Same with the _max()_ function `Expected math.max instead of max   scss/no-global-function-names` 
```scss
.foo {
  width: max(300px, 30%);
}
```

.stylelintrc.js
```js
module.exports = {
  'plugins': [
    'stylelint-scss',
  ],
  'rules': {
    'scss/no-global-function-names': true,
  },
};
```

```json
"stylelint": "13.7.0",
"stylelint-scss": "3.18.0"
```

https://caniuse.com/#feat=css-math-functions

**Comment by kristerkari** [comment]
Looks like the rule need to handle this as a special case. From Sass' docs:

> If a min() or max() function call is valid plain CSS, it will be compiled to a CSS min() or max() call. “Plain CSS” includes nested calls to calc(), env(), var(), min(), or max(), as well as interpolation. As soon as any part of the call contains a SassScript feature like variables or function calls, though, it’s parsed as a call to Sass’s core min() or max() function instead.

https://sass-lang.com/documentation/syntax/special-functions#min-and-max

**Comment by kristerkari** [comment]
Or is this a bug? Does it get solved by changing to use `math.min` and `math.max`, in that case the rule is working correctly.

**Comment by geebru** [comment]
@kristerkari Just confirming that it is a bug in the linter - seems like it needs to check for the values inside `min`, `max`, `clamp`, etc. for anything Sass-specific (variables, functions, etc) and validate against that?

As it stands it's returning errors for valid usages of `max` as a CSS function (not Sass).

**Comment by Donov4n** [comment]
Hi everyone :) Any known workaround for this ?

**Comment by kristerkari** [comment]
@Donov4n As a workaround you could add comments before and after your code to disable the rule for a single block of code, e.g.

```scss
// stylelint-disable scss/no-global-function-names
.foo {
  width: min(300px, 30%);
}
// stylelint-enable scss/no-global-function-names
```

More info: https://stylelint.io/user-guide/ignore-code/

**Comment by Donov4n** [comment]
Thank you for taking the time to answer me @kristerkari !

That's what I'm using now but what I meant was: Is there a more permanent solution than disabling the rule for now (so the answer is probably "no" given your answer).

I was asking this because it seems that the ticket hasn't received much interest in over a year and as it seems to be a fairly common problem I wondered if a solution had been found.

Thanks again !

**Comment by kristerkari** [comment]
I don't think that anything better can be done for now. 

From what I have understood, the rule would need to be fixed to detect when plain CSS min/max is used.

**Comment by johnc-ftl** [comment]
Please reopen this issue:

Using "stylelint-scss": "4.0.1", this bug is now fixed in most places where I was seeing it, but I'm still seeing a few of these errors. Eg:

```
 45:19  ✖  Expected math.min instead of min  scss/no-global-function-names
 54:18  ✖  Expected math.min instead of min  scss/no-global-function-names
```

For

```scss
      max-height: min(800px, 70vh);
      max-width: min(800px, 90vw);
```

I don't know why these aren't ignored, while all my other uses of min/max that were previously causing this lint error are now working fine. The only difference I can see is that they're within a few nested selectors.

**Comment by johncrim** [comment]
Please reopen this issue:

Using "stylelint-scss": "4.0.1", this bug is now fixed in most places where I was seeing it, but I'm still seeing a few of these errors. Eg:

```
 45:19  ✖  Expected math.min instead of min  scss/no-global-function-names
 54:18  ✖  Expected math.min instead of min  scss/no-global-function-names
```

For

```scss
      max-height: min(800px, 70vh);
      max-width: min(800px, 90vw);
```

I don't know why these aren't ignored, while all my other uses of min/max that were previously causing this lint error are now working fine. The only difference I can see is that they're within a few nested selectors.

**Comment by mplayer78** [comment]
For anyone looking at this thread, I have found that the easiest way to "fix" this issue is to use MAX, MIN, CALC, etc for css functions which clash, you may have to override another lowercase rule but this feels neater that commenting out lines of css.

**Comment by mkrause** [comment]
I'm hitting this issue with `round()`.
